NumLib/DOF/DOFTableUtil.cpp

```diff
@@ -14,33 +14,101 @@ namespace NumLib
 {
 namespace
 {
-
-template <class CalculateNorm>
+template <typename CalculateNorm>
 double norm(GlobalVector const& x, unsigned const global_component,
-             LocalToGlobalIndexMap const& dof_table, MeshLib::Mesh const& mesh, CalculateNorm calculate_norm)
+            LocalToGlobalIndexMap const& dof_table, MeshLib::Mesh const& mesh,
+            CalculateNorm calculate_norm)
 {
-    // TODO that also includes ghost nodes.
+#ifdef USE_PETSC
+    x.setLocalAccessibleVector();
+#endif
+
     double res = 0.0;
```

Comment:
I think it is clear that the result of the `norm` function is the 'norm'. I would prefer to leave it as it is.